clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/CloneOpenstackAtomicOperation.groovy

```diff
@@ -63,23 +64,23 @@ class CloneOpenstackAtomicOperation implements AtomicOperation<DeploymentResult>
 
     task.updateStatus BASE_PHASE, "Reading ancestor stack name ${description.source.stackName}..."
 
-    HeatStack ancestorStack = description.credentials.provider.getServerGroup(description.source.region, description.source.stackName)
-    if (!ancestorStack) {
-      throw new OpenstackResourceNotFoundException(AtomicOperations.CLONE_SERVER_GROUP, "Source stack ${description.source.stackName} does not exist.")
+    try {
+      HeatStack ancestorStack = description.credentials.provider.getServerGroup(description.source.region, description.source.stackName)
+      def ancestorNames = Names.parseName(description.source.stackName)
+
+      // Build description of object from ancestor, override any values that were specified on the clone call
+      newDescription.application = description.application ?: ancestorNames.app
+      newDescription.stack = description.stack ?: ancestorNames.stack
+      newDescription.freeFormDetails = description.freeFormDetails ?: ancestorNames.detail
+      newDescription.region = description.region ?: description.source.region
+      newDescription.heatTemplate = description.heatTemplate ?: description.credentials.provider.getHeatTemplate(description.source.region, ancestorStack.name, ancestorStack.id)
+      newDescription.parameters = description.parameters ?: [:]
```

Comment:
I was just updating the error handling here, I didn't pay attention to that line. I can change it though cause that makes more sense.